packages/lab/src/drawer/Drawer.tsx

```diff
@@ -59,52 +68,67 @@ export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(function Drawer(
   });
 
   const [showComponent, setShowComponent] = useState(false);
+  const { Component: FloatingComponent } = useFloatingComponent();
 
-  const { floating, context } = useDrawer({
+  const id = useId(idProp);
+
+  const { context, floating } = useFloatingUI({
     open,
     onOpenChange,
   });
 
-  const floatingRef = useForkRef<HTMLDivElement>(floating, ref);
+  const { getFloatingProps } = useInteractions([
+    useRole(context, { role: "dialog" }),
+    useClick(context),
+    useDismiss(context),
+  ]);
+
+  const handleRef = useForkRef<HTMLDivElement>(floating, ref);
+
+  const getDrawerProps = () => {
+    return getFloatingProps({
+      ref: floating,
+      id: `${id}-drawer`,
+    });
+  };
 
   useEffect(() => {
     if (open && !showComponent) {
       setShowComponent(true);
     }
-  }, [open, showComponent]);
+  }, [open, showComponent, setShowComponent]);
+
+  if (!showComponent) return null;
```

Comment:
Same comment as Dialog (https://github.com/jpmorganchase/salt-ds/pull/2958#discussion_r1472755033). How does this affect the Desktop support if FloatingComponent's open prop is not used as the source of truth?

Reply:
think this should be changed to a conditional render so floating component is mounted in the dom { showComponent && <Scrim> ... }